src/NuGet.Services.EndToEnd/Support/TestSettings.cs

```diff
@@ -28,86 +31,148 @@ public enum Mode
-                        searchBaseUrl: null,
-                        semVer2Enabled: false,
-                        searchInstanceCount: 2);
                 case Mode.Prod:
-                    return new TestSettings(
-                        DefaultAggressivePush,
-                        "https://www.nuget.org",
-                        "https://api.nuget.org/v3/index.json",
-                        new List<string>(),
-                        "API_KEY",
-                        searchBaseUrl: null,
-                        semVer2Enabled: false,
-                        searchInstanceCount: 5);
+                    {
+                        return DefaultTestSettings[mode];
+                    }
                 default:
-                    return new TestSettings(
-                        DefaultAggressivePush,
-                        EnvironmentSettings.GalleryBaseUrl,
-                        EnvironmentSettings.V3IndexUrl,
-                        EnvironmentSettings.TrustedHttpsCertificates,
-                        EnvironmentSettings.ApiKey,
-                        EnvironmentSettings.SearchBaseUrl,
-                        EnvironmentSettings.SemVer2Enabled,
-                        EnvironmentSettings.SearchInstanceCount);
+                    {
+                        return new TestSettings(
+                            DefaultAggressivePush,
+                            EnvironmentSettings.GalleryBaseUrl,
+                            EnvironmentSettings.V3IndexUrl,
+                            EnvironmentSettings.TrustedHttpsCertificates,
+                            EnvironmentSettings.ApiKey,
+                            EnvironmentSettings.SemVer2Enabled,
+                            GetAzureManagementConfiguration(secretReader),
+                            EnvironmentSettings.Subscription,
+                            EnvironmentSettings.SearchServiceResourceGroup,
+                            EnvironmentSettings.SearchServiceName);
+                    }
             }
         }
+
+        private static ISecretReader GetSecretReader()
+        {
+            string vaultName = EnvironmentSettings.KeyVaultName;
+
+            if (!string.IsNullOrEmpty(vaultName))
+            {
+                string clientId = EnvironmentSettings.KeyVaultClientId;
+                string certificateThumbprint = EnvironmentSettings.KeyVaultCertificateThumbprint;
+
+                var certificate = CertificateUtility.FindCertificateByThumbprint(StoreName.My, StoreLocation.LocalMachine, certificateThumbprint, validationRequired: false);
+
+                var keyVaultConfiguration = new KeyVaultConfiguration(vaultName, clientId, certificate);
+
+                return new CachingSecretReader(new KeyVaultReader(keyVaultConfiguration));
+            }
+
+            return new EmptySecretReader();
+        }
+
+        private static IAzureManagementAPIWrapperConfiguration GetAzureManagementConfiguration(ISecretReader secretReader)
```

Comment:
Helper methods should always be `async` make the sync caller call `.Result` or `.Wait`.